modules/tfm/tfm/boards/board/device_cfg.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2016-2019 ARM Limited
+ *
+ * Licensed under the Apache License Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ARM_LTD_DEVICE_CFG_H__
+#define __ARM_LTD_DEVICE_CFG_H__
+
+/**
+ * \file device_cfg.h
+ * \brief
+ * This is the default device configuration file with all peripherals
+ * defined and configured to be use via the secure and/or non-secure base
+ * address.
+ */
+
+#include <autoconf.h>
```

Comment:
Is this include necessary?

Reply:
It is required in order to include devicetree.h without errors.

Reply:
discussed offline

Reply:
I've made an attempt at fixing this upstream.

Header files should include their dependencies.

https://github.com/zephyrproject-rtos/zephyr/pull/50852

If this is not accepted we might want to eventually look into gcc -include autoconf.h from TF-M's build system.

Reply:
I think you are going against a principle in the Zephyr build system, since they explicitly use `-imacros` to avoid the problem of not including the configuration header.

In the TF-M build system it would probably be a problem to include the autoconf.h header all of the source files.
So including it only in the source files where we want to re-use the zephyr definitions we are reducing the chances of problems.

We can look into alternative solutions for how to improve this outside of this PR.